util/vendor.py

```diff
@@ -543,6 +572,22 @@ def git_add_commit(paths, commit_msg):
         log.warning("Unable to create commit. Are there no changes?")
 
 
+def define_arg_type(arg):
+    """Sanity-check and return a config file override argument"""
+    try:
+        (key, value) = [v.strip() for v in arg.split('=', 2)]
+    except Exception:
```

Comment:
Sorry, I should have noticed this before. Should this be a `TypeError`, rather than a bare `Exception`?

Reply:
This piece of code can give us a variety of different errors. At least `ValueError` if the split fails, `AttributeError` if `arg` happens to be something without a `split()` method, and probably a couple others I forgot.  Since the error handling will be mostly the same anyways I feel the catch-all approach is sufficient. Let me know if you'd like to go for a more targeted approach here.

Reply:
Well, `arg` is always a string when this is called by argparse, and `split` never fails on a string (e.g. `'abc'.split('=')` is `['abc']`). If `arg` isn't a string, or something else goes wrong, this is a programming error in the code and I would imagine we want it to go bang, rather than print out an (unrelated) error message.

But, as I said, it doesn't seem like a big deal either way.

Reply:
A good show-case of how easy it is to miss exception causes :smile: 

```python3
>>> arg="something"
>>> (key, value) = [v.strip() for v in arg.split('=', 2)]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: not enough values to unpack (expected 2, got 1)
```

I'll leave it as-is for now, until I get annoyed the next time it swallows a programming error and decide to go the other way :smile: